modules/gapi/src/backends/common/gbackend.hpp

```diff
@@ -58,31 +58,31 @@ namespace gimpl {
     struct Data;
     struct RcDesc;
 
-    struct GAPI_EXPORTS RMatMediaAdapterBGR final: public cv::RMat::Adapter
+    struct GAPI_EXPORTS RMatMediaFrameAdapter final: public cv::RMat::Adapter
     {
-        explicit RMatMediaAdapterBGR(const cv::MediaFrame& frame) : m_frame(frame) { };
+        explicit RMatMediaFrameAdapter(const cv::Mat& m, const cv::MediaFrame& f) :
+            m_mat(m),
+            m_frame(f)
+        { };
 
         virtual cv::RMat::View access(cv::RMat::Access a) override
         {
             auto view = m_frame.access(a == cv::RMat::Access::W ? cv::MediaFrame::Access::W
                                                                 : cv::MediaFrame::Access::R);
-            auto ptr = reinterpret_cast<uchar*>(view.ptr[0]);
-            auto stride = view.stride[0];
 
             std::shared_ptr<cv::MediaFrame::View> view_ptr =
                 std::make_shared<cv::MediaFrame::View>(std::move(view));
             auto callback = [view_ptr]() mutable { view_ptr.reset(); };
 
-            return cv::RMat::View(desc(), ptr, stride, callback);
+            return asView(m_mat, callback);
```

Comment:
1) In that line `m_mat` can already reference to the deallocated memory. (due `view` is destructed)
2) What's the point to take `view` if you already have `m_mat` ?